adapt_med_seg/data/dataset.py

```diff
@@ -231,13 +232,20 @@ def _load_single_dataset(self, data_json: str, splits: List[str|Tuple[str,str]])
 
     def get_train_val_dataloaders(
         self,
-        batch_size: int = 1
+        batch_size: int = 1,
+        max_len_samples: int = None
     ) -> tuple[DataLoader, DataLoader]:
         if not self.train:
             raise ValueError("This method is only for training dataset")
 
         train_subset = self._tr_val_splits["training"]
         val_subset = self._tr_val_splits["validation"]
+        if (max_len_samples is not None
+            and max_len_samples < len(train_subset)
+            and max_len_samples < len(val_subset)):
```

Comment:
This part here might be too restricting, because if the val set is small then this condition may not be true in many cases. Depends on what are the use cases

Reply:
lmk if I should fix it

Reply:
then the get_test_dataloader calls should not pass this argument. Leaving it like this for now but keep in mind